frontend/src/components/associations/marketplace/Orders.tsx

```diff
@@ -0,0 +1,170 @@
+import { PageTitle } from "../../utils/PageTitle";
+import Card from "react-bootstrap/Card";
+import { Pagination } from "../../utils/Pagination";
+import { Table } from "../../utils/table/Table";
+import { api } from "../../../services/apiService";
+import {
+    Transaction,
+    TransactionStatus,
+} from "../../../models/associations/marketplace";
+import { OverlayTrigger, Tooltip, Overlay } from "react-bootstrap";
+import { ToastContext, ToastLevel } from "../../utils/Toast";
+import { formatDate } from "../../../utils/format";
+
+export const AssociationMarketplaceOrders = ({ association }) => {
+    const marketplaceId = association.id;
+    const columns = [
+        {
+            key: "product",
+            header: "Produit",
+            render: (transaction) => transaction.product.name,
+        },
+        {
+            key: "quantity",
+            header: "Quantité",
+            headerClassName: "text-center",
+            cellClassName: "text-center",
+        },
+        {
+            key: "date",
+            header: "Date",
+            render: (transaction) => formatDate(new Date(transaction.date)),
+            headerClassName: "text-center",
+            cellClassName: "text-center",
+        },
+        {
+            key: "value",
+            header: "Valeur",
+            headerClassName: "text-right",
+            cellClassName: "text-right",
+            render: (transaction) => transaction.value + "€",
+        },
+        {
+            key: "buyer",
+            header: "Utilisateur",
+            render: (transaction) => transaction.buyer,
+            headerClassName: "text-center",
+            cellClassName: "text-center",
+        },
+        {
+            key: "status",
+            header: "Status",
+            render: (transaction) => (
+                <TransactionStatusSelector transaction={transaction} />
+            ),
+        },
+    ];
+
+    return (
+        <>
+            <PageTitle>Commandes</PageTitle>
+
+            <Pagination
+                render={(transactions: Transaction[], paginationControl) => (
+                    <>
+                        <Card>
+                            <Table columns={columns} data={transactions} />
+                        </Card>
+                        {paginationControl}
+                    </>
+                )}
+                apiKey={[
+                    "transactions.list.admin",
+                    marketplaceId,
+                    { page_size: 10, ordering: "-date" },
+                ]}
+                apiMethod={api.transactions.list}
+                config={{ refetchOnWindowFocus: false }}
+                paginationControlProps={{
+                    className: "justify-content-center mt-5",
+                }}
+            />
+        </>
+    );
+};
+
+const TransactionStatusSelector = ({
+    transaction,
+}: {
+    transaction: Transaction;
+}) => {
+    let statusList = [
```

Comment:
`const`?

Reply:
Also, I feel like this could be used somewhere else in the component, and should thus be near `TransactionStatus`.